File: 5_multiple_inheritance.py
class Shape:
    def __init__(self, width, height):
        self.width = width
        self.height = height

class Color:
    def __init__(self, color):
        self.color = color

class Rectangle(Shape, Color):
    def __init__(self, width, height, color):
        Shape.__init__(self, width, height)
        Color.__init__(self, color)

    def display(self):
        print(f"The rectangle Width is : {self.width}, Height: {self.height}, Color: {self.color}")


rect = Rectangle(5, 3, "Orange")

rect.display()

File: 5_multiple_inheritance.cpp
#include <bits/stdc++.h>
using namespace std;

class Shape
{
protected:
    int width;
    int height;

public:
    Shape(int w, int h)
    {
        width = w;
        height = h;
    }
};

class Color
{
protected:
    string color;

public:
    Color(string c) : color(c) {}
};

class Rectangle : public Shape, public Color
{
public:
    Rectangle(int w, int h, string c) : Shape(w, h), Color(c) {}

    void display()
    {
        cout <<"The rectangle "<< "Width: " << width << ", Height: " << height << ", Color: " << color << endl;
    }
};

int main()
{
    Rectangle rectangle(5, 3, "Orange");

    rectangle.display();

    return 0;
}
